server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -331,8 +345,16 @@ public Boolean requestCache() {
         return requestCache;
     }
 
-    public void requestCache(Boolean requestCache) {
-        this.requestCache = requestCache;
+    public BytesReference cacheModifier() {
+        return cacheModifier;
+    }
+
+    /**
+     * The cache modifier distinguishes between requests that are otherwise identical, but should be treated as independent for caching
+     * purposes - typically because there is some reason that the results will be different, even though they execute the same query
+     */
+    public void cacheModifier(BytesReference cacheModifier) {
```

Comment:
Thanks Jim. Plugin extension is indeed an alternative to achieve the goal. It makes the whole thing more formal. It could also be a bigger change and hence is better justified if the extension point could be reused. Do you have idea about other places this extension could become useful? Nevertheless, I'll expore this alternative to get a better understanding on how it would work.

Reply:
I cannot think of other places where this extension would be useful. However my point is on how we implement the feature. It seems more natural to use an extension point even if it's only for security. Overloading the shard request seems a bit hacky imo.